sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -97,6 +97,14 @@ object DateTimeUtils {
     millisLocal - getOffsetFromLocalMillis(millisLocal, timeZone)
   }
 
+  def microsToDays(epochMicros: SQLTimestamp): SQLDate = {
+    Math.floorDiv(epochMicros, MICROS_PER_DAY).toInt
```

Comment:
this one is a little tricky. There is a comment nearby, saying that `Math.floorDiv` is necessary because we want to support days before UTC epoch. Does `MICROSECONDS.toDay` handle negative values?

Reply:
> Does MICROSECONDS.toDay handle negative values?

Don't think so. It just do div: `public long toDays(long d)    { return d/(C6/C1); }`

Reply:
maybe we should send a PR to open JDK :P